antispam/management/commands/update_spam_database.py

```diff
@@ -0,0 +1,56 @@
+import requests
+from django.core.management import BaseCommand
+from django.utils import timezone
+
+from antispam.models import ToxicDomain
+
+
+class Command(BaseCommand):
+    """Update blocked ips/mails database"""
+
+    help = "Update blocked ips/mails database"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--force", action="store_true", help="Force re-creation even if up to date"
+        )
+
+    def _should_update(self, cls: type[ToxicDomain], *, force: bool = False) -> bool:
+        if force:
+            return True
+        oldest = (
+            cls.objects.filter(is_externally_managed=True).order_by("created").first()
+        )
+        return not (
+            oldest and timezone.now() < oldest.created + timezone.timedelta(days=1)
+        )
+
+    def _update_domains(self, *, force: bool = False):
+        source = "https://www.stopforumspam.com/downloads/toxic_domains_whole.txt"
+        if not self._should_update(ToxicDomain, force=force):
+            self.stdout.write("Domain database is up to date")
+            return
+        self.stdout.write("Updating domain database")
+
+        resp = requests.get(source)
+        if resp.status_code != 200:
+            raise requests.HTTPError(
+                f"Toxic domain update failed with error {resp.status_code}"
+            )
```

Comment:
```suggestion
        if not self._should_update(force=force):
            self.stdout.write("Domain database is up to date")
            return
        self.stdout.write("Updating domain database")

        sources = settings.TOXIC_DOMAINS_PROVIDERS
        domains = {}
        for source in sources:
            res = request.get(source)
            if not res.ok:
                self.stderr.write(f"Source {source} responded with code {res.status_code}")
                continue
            domains |= set(res.content.decode().splitlines())
```
Ca veut dire qu'il faut ajouter `TOXIC_DOMAINS_PROVIDERS` dans les settings